src/runtime/pkg/kata-monitor/pprof.go

```diff
@@ -73,38 +82,64 @@ func (km *KataMonitor) proxyRequest(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("Content-Disposition", contentDisposition)
 	}
 
-	io.Copy(w, output)
+	proxyResponse(r, w, output)
 }
 
 // ExpvarHandler handles other `/debug/vars` requests
 func (km *KataMonitor) ExpvarHandler(w http.ResponseWriter, r *http.Request) {
-	km.proxyRequest(w, r)
+	km.proxyRequest(w, r, nil)
 }
 
 // PprofIndex handles other `/debug/pprof/` requests
 func (km *KataMonitor) PprofIndex(w http.ResponseWriter, r *http.Request) {
-	km.proxyRequest(w, r)
+	if len(strings.TrimPrefix(r.URL.Path, "/debug/pprof/")) == 0 {
+		km.proxyRequest(w, r, copyResponseAddingSandboxIdToHref)
+	} else {
+		km.proxyRequest(w, r, nil)
+	}
 }
 
 // PprofCmdline handles other `/debug/cmdline` requests
 func (km *KataMonitor) PprofCmdline(w http.ResponseWriter, r *http.Request) {
-	km.proxyRequest(w, r)
+	km.proxyRequest(w, r, nil)
 }
 
 // PprofProfile handles other `/debug/profile` requests
 func (km *KataMonitor) PprofProfile(w http.ResponseWriter, r *http.Request) {
-	km.proxyRequest(w, r)
+	km.proxyRequest(w, r, nil)
 }
 
 // PprofSymbol handles other `/debug/symbol` requests
 func (km *KataMonitor) PprofSymbol(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
-	km.proxyRequest(w, r)
+	km.proxyRequest(w, r, nil)
 }
 
 // PprofTrace handles other `/debug/trace` requests
 func (km *KataMonitor) PprofTrace(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/octet-stream")
 	w.Header().Set("Content-Disposition", `attachment; filename="trace"`)
-	km.proxyRequest(w, r)
+	km.proxyRequest(w, r, nil)
+}
+
+func copyResponse(req *http.Request, w io.Writer, r io.Reader) {
+	io.Copy(w, r)
+}
+
+func copyResponseAddingSandboxIdToHref(req *http.Request, w io.Writer, r io.Reader) {
+	sb, err := getSandboxIDFromReq(req)
+	if err != nil {
+		monitorLog.WithError(err).Warning("missing sandbox query in pprof url")
+		copyResponse(req, w, r)
+		return
+	}
+	buf, err := io.ReadAll(r)
+	if err != nil {
+		monitorLog.WithError(err).Fatal("cannot read pprof index page")
+		os.Exit(1)
```

Comment:
Don't `os.Exit`, use `serveError() && return` instead, `os.Eixt` will terminate kata-monitor process.

Reply:
mmm, yeah, maybe too drastic to terminate kata-monitor :-)
Reworked to return an error (opted for an "InternalServer" error as my guess would be something weird happened with the connection to the runtime), how that looks?